osmosis-labs/osmojs
# e2e testing with Starship

Closes #22 

NOTE: While merging only Squash and Merge this PR (PR has commits for debugging CI/CD, which should not be part of the main commit history)

## Overview
Create a multi-chain ecosystem, against which the tests are run.

## How to run the tests
* `cd osmojs/packages/osmojs/tests`
* Connect to a k8s cluster or create it with Kind or connect to one, `kubectl get pods` should work
* Run `make setup` to get the helm charts
* Run `make install` to spin up the mini-cosmos ecosystem
* Optionally: Install and run `k9s` to interact with the k8s cluster
* Once everything is in running stage (time to spin up will depend on the resources constraints of the machine)
* Run `make port-forward` to be able to interact with the system

Then once you have the infra working run the jest tests
`yarn run e2e:test`
